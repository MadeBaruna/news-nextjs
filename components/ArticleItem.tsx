import React, { StatelessComponent } from 'react';
import styled from 'styled-components';

interface IProps {
  article: Article;
}

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  padding: 8px;
  border-bottom: 1px solid #ddd;

  @media screen and (max-width: 699px) {
    flex-direction: column-reverse;
  }
`;

const TextWrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin-right: 10px;
  flex: 1;
`;

const Title = styled.h4`
  margin: 0;
  font-weight: 700;
  font-family: "Raleway", sans-serif;
  margin-bottom: 5px;
  text-decoration: none;
  color: black;
`;

const Description = styled.p`
  font-family: "Open Sans", sans-serif;
  font-size: 14px;
  margin: 0;
  color: #777;
  text-decoration: none;
`;

const ImageWrapper = styled.div`
  width: 200px;
  height: 112.5px;

  @media screen and (max-width: 699px) {
    width: 100%;
    height: auto;
    max-height: 300px;
  }
`;

const Image = styled.img`
  width: 200px;
  height: 112.5px;
  object-fit: cover;

  @media screen and (max-width: 699px) {
    width: 100%;
    height: auto;
    max-height: 300px;
  }
`;

const Link = styled.a`
  text-decoration: none;
`;

const ArticleItem: StatelessComponent<IProps> = ({ article }) => (
  <Link target="__blank" href={article.url}>
    <Wrapper>
      <TextWrapper>
        <Title>{article.title}</Title>
        <Description>{article.description}</Description>
      </TextWrapper>
      {article.urlToImage && (
        <ImageWrapper>
          <Image src={article.urlToImage} />
        </ImageWrapper>
      )}
    </Wrapper>
  </Link>
);

export default ArticleItem;
