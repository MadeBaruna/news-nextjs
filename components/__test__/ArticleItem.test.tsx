import { shallow } from 'enzyme';
import React from 'react';
import ArticleItem from '../ArticleItem';

describe('Article Item', () => {
  it('Render Article Item', () => {
    const article = {
      source: {
        id: null,
        name: 'Thetimes.co.uk',
      },
      author: null,
      title:
        'Growing up near green spaces is linked to better mental health - The Times',
      description:
        'Children who grow up in greener surroundings have a greatly reduced risk of developing mental illnesses later in life, research suggests. A study that tracked almost a million people found that...',
      url:
        'https://www.thetimes.co.uk/article/growing-up-near-green-spaces-is-linked-to-better-mental-health-hgrrhlgxv',
      urlToImage:
        'http://www.thetimes.co.uk/imageserver/image/methode%2Ftimes%2Fprod%2Fweb%2Fbin%2F2144c664-3946-11e9-a664-d81d3e2c8975.jpg?crop=2222%2C1250%2C0%2C116',
      publishedAt: '2019-02-26T00:01:00Z',
      content:
        'Children who grow up in greener surroundings have a greatly reduced risk of developing mental illnesses later in life, research suggests. A study that tracked almost a million people found that those who were raised among the lowest levels of green space wereâ€¦ [+480 chars]',
    };

    const wrapper = shallow(<ArticleItem article={article} />);
    expect(wrapper).toMatchSnapshot();
  });
});
