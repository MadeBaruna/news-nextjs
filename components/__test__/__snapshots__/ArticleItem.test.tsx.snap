// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Item Render Article Item 1`] = `ShallowWrapper {}`;
